addons/account/tests/test_early_payment_discount.py

```diff
@@ -61,6 +61,19 @@ def test_early_payment_end_date(self):
                     fields.Date.from_string('2019-01-11') or False
                 )
 
+    def test_invoice_report_without_invoice_date(self):
+        #invoice with payment term of early discount
+        out_invoice = self.env['account.move'].create([{
+            'move_type': 'out_invoice',
+            'invoice_payment_term_id': self.early_pay_10_percents_10_days.id,
+            'invoice_line_ids': [Command.create({
+                'name': 'line1',
+            })]
+        }])
+
+        report = self.env['ir.actions.report'].with_context(force_report_rendering=True)._render_qweb_pdf('account.account_invoices', res_ids=out_invoice.id)
+        self.assertTrue(report)
```

Comment:
Could you please add an assert to verify that invoice date is not set and explain in a comment what you're testing here? 

Thank you